src/librustc/ty/mod.rs

```diff
@@ -257,6 +264,130 @@ impl AssocItem {
     }
 }
 
+/// A list of `ty::AssocItem`s in definition order that allows for efficient lookup by name.
+///
+/// When doing lookup by name, we try to postpone hygienic comparison for as long as possible since
+/// it is relatively expensive. Instead, items are indexed by `Symbol` and hygienic comparison is
+/// done only on items with the same name.
+#[derive(Debug, Clone, PartialEq, HashStable)]
+pub struct AssociatedItems {
+    items_in_def_order: Vec<ty::AssocItem>,
+    item_idx_in_name_order: Vec<usize>,
```

Comment:
This commit contains a chunk of other removed commit.

Reply:
Whoops. My bad.